### DISCUSSION TITLE
Render Hooks for Images in Markdown

### DISCUSSION THREAD

**Discussion by zivbk1** [pngwn/MDsveX#438] [Q&A] [answered]
Hello,

I come from Hugo, and there is a feature called 'Render Hooks' that allows you to specify how things like Headings, Images, Links, etc. are rendered to HTML when the markdown processor encounters them.  You can see an example of this in the Hugo docs here: https://gohugo.io/templates/render-hooks/#image-markdown-example

I am looking for an equivalent in MDsveX.  Would I use a remark/rehype plugin, or is there another way to add some logic and special formatting for images from my markdown files?

Thank you,
- Bryan

**Comment by pngwn** [comment, marked as answer]
A `rehype` plugin would be the equivalent to these `renderhooks`. You get access to an AST and can transform that structure as you wish. 1.0 will have a simpler mechanism for the kind of thing.

Alternatively you could use a custom component which would be simpler: https://mdsvex.pngwn.io/docs#custom-components
